clippy_lints/src/methods/mod.rs

```diff
@@ -1915,15 +1916,51 @@ fn lint_wrong_self_convention<'tcx>(
     } else {
         WRONG_SELF_CONVENTION
     };
-    if let Some((ref conv, self_kinds)) = &CONVENTIONS.iter().find(|(ref conv, _)| conv.check(item_name)) {
+    if let Some((conventions, self_kinds)) = &CONVENTIONS
+        .iter()
+        .find(|(convs, _)| convs.iter().all(|conv| conv.check(item_name)))
+    {
         if !self_kinds.iter().any(|k| k.matches(cx, self_ty, first_arg_ty)) {
+            let suggestion = {
+                if conventions.len() > 1 {
+                    let special_case = {
+                        // Don't mention `NotEndsWith` when there is also `StartsWith` convention present
+                        if conventions.len() == 2 {
+                            match conventions {
+                                [Convention::StartsWith(starts_with), Convention::NotEndsWith(_)]
+                                | [Convention::NotEndsWith(_), Convention::StartsWith(starts_with)] => {
+                                    Some(format!("methods called `{}`", Convention::StartsWith(starts_with)))
+                                },
+                                _ => None,
+                            }
+                        } else {
+                            None
+                        }
+                    };
+                    match special_case {
+                        Some(suggestion) => suggestion,
+                        None => {
+                            let s = conventions
+                                .iter()
+                                .map(|c| format!("`{}`", &c.to_string()))
+                                .collect::<Vec<_>>()
+                                .join(" and ");
+
+                            format!("methods called like this: ({})", &s)
+                        },
+                    }
+                } else {
+                    format!("methods called `{}`", &conventions[0])
+                }
+            };
+
             span_lint(
                 cx,
                 lint,
                 first_arg_span,
                 &format!(
-                    "methods called `{}` usually take {}; consider choosing a less ambiguous name",
-                    conv,
+                    "{} usually take {}; consider choosing a less ambiguous name",
```

Comment:
While your at it: Could you change this to `span_lint_and_help` please? The `consider choosing a less ambiguous name` part should not be in the error message, but in a help message.